ts/demo-functions/src/helm_template.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { safeLoadAll } from 'js-yaml';
+import {
+  Configs,
+  FunctionConfigError,
+  isKubernetesObject,
+  ConfigError,
+} from 'kpt-functions';
+import { spawnSync } from 'child_process';
+import { isConfigMap } from './gen/io.k8s.api.core.v1';
+
+const NAME = 'name';
+const CHART_PATH = 'chart_path';
+const VALUES_PATH = '--values';
+
+// Render chart templates locally using helm template.
+export async function helmTemplate(configs: Configs) {
+  // Validate config data and read arguments.
+  const args = readArguments(configs);
+  args.unshift('template');
+
+  let child;
```

Comment:
why is these not defined when assigned?

Reply:
Defining child when assigned now. We need error to have function scope so it is still defined without assignment.